ServiceManager/GUI/main_window.py

```diff
@@ -0,0 +1,629 @@
+        self.update_config_data()
+        self.update_config_table()
+
+    def show_filter_btn_clicked(self):
+        """ Show or Hide the Filters bar. """
+        toggle = not self.filter_frame.isVisible()
+        self.filter_frame.setVisible(toggle)
+
+    def apply_filters(self):
+        """
+        Apply filters and display only the matches in the config table.
+        column_of_interest = 0 -> index 0 for the filters columns comboBox (All Columns).
+         """
+        column_of_interest = self.filter_columns_cb.currentIndex()
+        value_of_interest = self.filter_bar.text()
+
+        if column_of_interest == 0:
+            results = self.config_table.findItems(value_of_interest, Qt.MatchContains)
+            rows = set(item.row() for item in results)
+            for index in range(self.config_table.rowCount()):
+                show_or_hide = index not in rows
+                self.config_table.setRowHidden(index, show_or_hide)
+        else:
+            column_of_interest -= 1  # So indexes match (as filters columns comboBox has an extra "All Columns" on 0)
+            for rowIndex in range(self.config_table.rowCount()):
+                item = self.config_table.item(rowIndex, column_of_interest)
+                if value_of_interest in item.text():
+                    self.config_table.setRowHidden(rowIndex, False)
+                else:
+                    self.config_table.setRowHidden(rowIndex, True)
+
+    def clear_filters(self):
+        self.filter_columns_cb.setCurrentIndex(0)
+        self.filter_bar.clear()
+        self.refresh_config()
+
+    def new_config_btn_clicked(self):
+        """ Open the Config Entry dialog window. If database is not connected, display error message. """
+        if not DBUtils.is_connected():
+            QMessageBox.critical(self, 'Database connection required',
+                                       'Database connection is required to edit the PV configuration.',
+                                       QMessageBox.Ok)
+            return
+
+        if self.config_entry_w is None:
+            self.config_entry_w = UIConfigEntryDialog()
+            self.config_entry_w.config_updated.connect(self.refresh_config)
+        self.config_entry_w.show()
+        self.config_entry_w.activateWindow()
+
+    def edit_config_btn_clicked(self):
+        """ On Edit, open Config Entry dialog window as normal with object selected and PV config loaded. """
+        if not DBUtils.is_connected():
+            QMessageBox.critical(self, 'Database connection required',
+                                       'Database connection is required to edit the PV configuration.',
+                                       QMessageBox.Ok)
+            return
+        if self.config_entry_w is None:
+            self.config_entry_w = UIConfigEntryDialog()
+            self.config_entry_w.config_updated.connect(self.refresh_config)
+        self.config_entry_w.show()
```

Comment:
**should**: This does the same as the above method, they should instead use a common method